include/jemalloc/internal/prof_structs.h

```diff
@@ -97,10 +104,10 @@ struct prof_tctx_s {
 typedef rb_tree(prof_tctx_t) prof_tctx_tree_t;
 
 struct prof_info_s {
-	/* Points to the prof_tctx_t corresponding to the allocation. */
-	prof_tctx_t		*alloc_tctx;
 	/* Time when the allocation was made. */
 	nstime_t		alloc_time;
```

Comment:
Question: Since now prof_recent tracks alloc_time, can we later remove this one here?

Reply:
I don't think so. The reason is as follows:

The `prof_info` design is mainly for cases like `rallocx()`, where the recording of profiling information takes place at a time the allocation might have been gone. Therefore, the `alloc_time` field on `prof_info` can be removed only if it's possible to locate the `prof_recent_alloc` record at the time the allocation is already gone. This is not possible in the current design: the only way to retrieve the recent allocation record given an allocation is the allocation-to-record pointer located on the allocation itself.